app/components/oauth/applications/row_component.rb

```diff
@@ -73,7 +85,29 @@ def edit_link
       end
 
       def button_links
-        [edit_link, helpers.delete_link(oauth_application_path(application))]
+        if application.builtin?
+          [toggle_enabled_link]
+        else
+          [edit_link, helpers.delete_link(oauth_application_path(application))]
+        end
+      end
+
+      def toggle_enabled_link
+        if application.enabled?
+          link_to(
+            I18n.t(:button_deactivate),
+            toggle_oauth_application_path(application),
+            class: "oauth-application--edit-link icon icon-lock",
+            method: :post
+          )
+        else
+          link_to(
+            I18n.t(:button_activate),
+            toggle_oauth_application_path(application),
+            class: "oauth-application--edit-link icon icon-unlock",
+            method: :post
```

Comment:
This leads to a link without icon which is unusual for the table:

<img width="260" alt="image" src="https://github.com/opf/openproject/assets/617519/d12dc820-50c9-4b55-92d4-ca5f50825a3d">

AFAIK, the intend is to move away from link to button based actions completely but this might be for a different time.

Reply:
A somewhat unrelated question: Why not allow toggling the active status for all applications?

Reply:
Interesting idea. However, I struggle to find a use case for it. You either want an application or not. Such a toggle for not built-in application would only make sense if the associated tokens would also get toggled (and not deleted, which btw is missing in this PR). But I believe that is not worth the effort.

Reply:
This is deprecated, as the implementation will embrace primer UI with a toggle.